packages/backend/src/session/service.ts

```diff
@@ -0,0 +1,86 @@
+import { BaseService } from '../shared/baseService'
+import { uuid } from '../connector/core'
+import IORedis from 'ioredis'
+import { isSessionError, SessionError } from './errors'
+import { Session, SessionKey } from './util'
+
+export interface SessionService {
+  create(): Promise<SessionKey>
+  revoke(session: SessionOptions): void
+  refresh(session: SessionOptions): Promise<SessionKey | SessionError>
+  get(session: SessionOptions): Promise<Session | SessionError>
```

Comment:
What do you think about removing `SessionOptions` in favor of
```suggestion
  revoke(key: string): void
  refresh(key: string): Promise<SessionKey | SessionError>
  get(key: string): Promise<Session | SessionError>
```

Reply:
Don't all services have `options`? I wanted to make it consistent.

Reply:
Yeah, a lot do especially for `create`.
I was just thinking that since there's just one option here, to deal with it as the string.

Reply:
I don't have a strong opinion on that. I'm happy to change it if you think it makes more sense.